### PR TITLE
Support defining write constraints for newly created fields

### PR DESCRIPTION
**Pull request by simpkins**
This updates `make_field()` to honor a writeConstraint field, the same way that `make_register()` does.
I included a new unit test that verifies the behavior.

### CODE REVIEW

File: res/add_register/stm32l4x2.svd [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8" standalone="no"?>
```
Comment (simpkins) [inline res/add_register/stm32l4x2.svd:1]: Note that I copied this SVD file from the existing `res/add/` test.  If preferred I can remove the copy, and just have the patch.yaml file refer to it as `../add/stm3214x2.svd`

Comment (burrbull) [reply res/add_register/stm32l4x2.svd:1]: Yes. Don't make copies if it is not needed, please.

**Comment by burrbull** [comment]
Is is strange this was missed. Thanks.

**Comment by simpkins** [comment]
Updated the `add_register` test to just refer to the existing SVD file from the `add` test.

**Review by burrbull — APPROVED** [review]

**Pull request merged**